src/esp/gfx/CubeMap.h

```diff
@@ -36,7 +36,12 @@ class CubeMap {
     Depth = 1,
     // TODO: ObjectId
     // TODO: HDR color
+
+    // you will have to update it when new type is added
+    Count = 2,
   };
+  static const uint8_t numTextureTypes =
+      static_cast<uint8_t>(TextureType::Count);
```

Comment:
You don't need this, it's redundant. Use `uint8_t(TextureType::Count)` in the two places where it's needed instead.

Reply:
Yes, in terms of code, it is just 2 places.
But in terms of usage, I hope to avoid casting it every time even it is light operation.

Reply:
I made the comment because this extra variable made the code harder to understand for me -- I had to check what exactly it was, if it really matched what's in the enum, and what was the possibility for it to get out of sync, which then led to the other comment below.

So in my personal opinion just a congitive overhead -- the cast is *a bit* of extra typing but immediately clear to whoever reads the code, and the code gets read much more often than it gets written ;)